### PR TITLE
01 tdd

### PR DESCRIPTION
**Pull request by gak92**
In this pull request:

- [X] Create test cases for the methods: factorial, reverse and fizzbuzz

### CODE REVIEW

File: src/spec/solver_spec.rb [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    it 'return 24 if parameter is 4' do
+      Solver.factorial(4).should eql 24
+    end
+  end
+
+  describe '#factorial' do
+    it 'raise an exception if parameter is negative' do
+      expect { Solver.factorial(-1) }.to raise_error('Number is negative')
+    end
+  end
```
Comment (GzimAsani) [inline src/spec/solver_spec.rb]: - [ ] This should raise an exception, kindly update it
```ruby
     ...
     describe Resolver do
         describe "#factorial" do
             it "raises exception for negative number" do
             resolver = Resolver.new
             expect { resolver.factorial(-5) }.to raise_exception
             end
         end
     end
```

**Review by GzimAsani — CHANGES_REQUESTED** [review]
### Hello @gak92 and @bienvenuushindi          :wave:   
 Good job so far! There are some issues that you still need to work on to go to the next project but you are almost there!
 
 ## Highlights
 * Good job with your tests, they are almost done:+1:  
 
 ## Required Changes 

- [ ] Check the comment below :arrow_double_down: 






 ## Optional suggestions
 _Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._



Cheers and Happy coding! clapclapclap
 
 Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.
 
 **_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**
 
 _As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by rotimiazeez — APPROVED** [review]
# Hi TEAM 👍🏽,
### Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by gak92** [comment]
@rotimiazeez,
Thanks for the review.

**Pull request merged**